src/filesystem/file_content.rs

```diff
@@ -97,6 +103,14 @@ impl FileContent {
         let mut written_bytes = 0;
 
         match self {
+            Self::EmptyFile => {
+                writer.write_all(&[FILE_CONTENTY_TYPE_EMPTY]).await?;
+                written_bytes += 1;
+
+                let data_size_bytes = 0u64.to_le_bytes();
+                writer.write_all(&data_size_bytes).await?;
+                written_bytes += data_size_bytes.len();
```

Comment:
For this node type we shouldn't encode the implicitly 0 as a data length. It works fine as long as both the parser and encoder are symmetrical (this isn't a format breaking change).